From 0fb6fcfb518a19b26c6ffc4773db05c093826da5 Mon Sep 17 00:00:00 2001
From: Mingkai Hu <Mingkai.Hu@freescale.com>
Date: Thu, 30 Apr 2015 18:39:47 +0800
Subject: [PATCH 106/451] arm64/ls1043a: do not slect GIC_V3 by default for
 ARM64

LS1043A use is ARMv8 core, but have GIC-400 with it.

Signed-off-by: Mingkai Hu <Mingkai.Hu@freescale.com>
[Xulin: Original patch taken from
Linux-LS1043A-SDK-V0.4-SOURCE-20150826-yocto.iso]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 arch/arm64/Kconfig |   10 ++++++++--
 1 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/Kconfig b/arch/arm64/Kconfig
index d9da51c..846e7fd 100644
--- a/arch/arm64/Kconfig
+++ b/arch/arm64/Kconfig
@@ -10,8 +10,6 @@ config ARM64
 	select ARM_AMBA
 	select ARM_ARCH_TIMER
 	select ARM_GIC
-	select ARM_GIC_V3
-	select ARM_GIC_V3_ITS if PCI_MSI
 	select AUDIT_ARCH_COMPAT_GENERIC
 	select BUILDTIME_EXTABLE_SORT
 	select CLONE_BACKWARDS
@@ -148,6 +146,8 @@ config LS2085A_ERRATA_ERR008585
 config ARCH_FSL_LS2085A
 	bool "Freescale LS2085A SOC"
 	select LS2085A_ERRATA_ERR008585
+	select ARM_GIC_V3
+	select ARM_GIC_V3_ITS if PCI_MSI
 	help
 	  This enables support for Freescale LS2085A SOC.
 
@@ -158,6 +158,8 @@ config ARCH_FSL_LS1043A
 
 config ARCH_THUNDER
 	bool "Cavium Inc. Thunder SoC Family"
+	select ARM_GIC_V3
+	select ARM_GIC_V3_ITS if PCI_MSI
 	help
 	  This enables support for Cavium's Thunder Family of SoCs.
 	select PCI
@@ -190,12 +192,16 @@ config ARCH_VEXPRESS
 	select COMMON_CLK_VERSATILE
 	select POWER_RESET_VEXPRESS
 	select VEXPRESS_CONFIG
+	select ARM_GIC_V3
+	select ARM_GIC_V3_ITS if PCI_MSI
 	help
 	  This enables support for the ARMv8 software model (Versatile
 	  Express).
 
 config ARCH_XGENE
 	bool "AppliedMicro X-Gene SOC Family"
+	select ARM_GIC_V3
+	select ARM_GIC_V3_ITS if PCI_MSI
 	help
 	  This enables support for AppliedMicro X-Gene SOC Family
 
-- 
1.7.5.4

